core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -893,6 +893,92 @@ const cudaDeviceProp &Cuda::cuda_device_prop() const {
   return m_space_instance->m_deviceProp;
 }
 
+namespace Impl {
+
+extern int get_gpu(const InitArguments &args);
```

Comment:
Why not just forward-declare normally?
```suggestion
int get_gpu(const InitArguments &args);
```